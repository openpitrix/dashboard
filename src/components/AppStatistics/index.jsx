import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { translate } from 'react-i18next';

import { Icon } from 'components/Base';
import { Card } from 'components/Layout';

import styles from './index.scss';

const stars = [1, 2, 3, 4, 5];

@translate()
export default class AppStatistics extends Component {
  static propTypes = {
    appTotal: PropTypes.number,
    className: PropTypes.string,
    isAppDetail: PropTypes.bool,
    monthDepoly: PropTypes.number,
    starTotal: PropTypes.number,
    totalDepoly: PropTypes.number,
    versionTotal: PropTypes.number
  };

  static defaultProps = {
    isAppDetail: false,
    appTotal: 0,
    versionTotal: 0,
    monthDepoly: 0,
    totalDepoly: 0,
    starTotal: 5
  };

  render() {
    const {
      className,
      isAppDetail,
      appTotal,
      versionTotal,
      monthDepoly,
      totalDepoly,
      starTotal,
      t
    } = this.props;

    return (
      <Card className={classnames(styles.appStatistics, className)}>
        <dl>
          <dt>{t(isAppDetail ? '线上版本' : '已上架应用')}</dt>
          <dd>{isAppDetail ? versionTotal : appTotal}</dd>
        </dl>
        <dl>
          <dt>{t('本月部署次数')}</dt>
          <dd>{monthDepoly}</dd>
        </dl>
        <dl>
          <dt>{t('总部署次数')}</dt>
          <dd>{totalDepoly}</dd>
        </dl>
        {/* <dl>
          <dt>{t('综合评价')}</dt>
          <dd>
            {starTotal}
            <label>
              {stars.map(star => (
                <Icon
                  key={star}
                  name="star"
                  size={16}
                  type="dark"
                  className={classnames({
                    [styles.yellow]: starTotal >= star
                  })}
                />
              ))}
            </label>
          </dd>
        </dl> */}
      </Card>
    );
  }
}
