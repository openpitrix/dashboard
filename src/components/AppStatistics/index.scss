@import '~scss/vars';

.appStatistics {
  padding: 32px 0  32px 50px;
  height: 104px;

  dl {
    float: left;
    display: inline-block;
    width: 25%;

    dt {
      @include note-font($line-height: 12px);
      margin-bottom: 12px;
    }

    dd {
      line-height: 16px;
      font-size: 18px;
    }

    .yellow svg {
      --primary-color: #{$Y75};
      --secondary-color: #{$Y75};
    }
  }
}
