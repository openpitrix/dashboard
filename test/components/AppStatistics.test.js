import AppStatistics from 'components/AppStatistics';

describe('AppStatistics', () => {
  it('basic render', () => {
    const wrapper = render(<AppStatistics />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('app detail statistics', () => {
    const wrapper = render(<AppStatistics isAppDetail />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
