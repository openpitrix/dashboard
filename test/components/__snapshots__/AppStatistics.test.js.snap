// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppStatistics app detail statistics 1`] = `
<div
  class="panel card appStatistics"
>
  <dl>
    <dt>
      Online version
    </dt>
    <dd>
      0
    </dd>
  </dl>
  <dl>
    <dt>
      Deploy times this month
    </dt>
    <dd>
      0
    </dd>
  </dl>
  <dl>
    <dt>
      Total of deploy
    </dt>
    <dd>
      0
    </dd>
  </dl>
</div>
`;

exports[`AppStatistics basic render 1`] = `
<div
  class="panel card appStatistics"
>
  <dl>
    <dt>
      On the shelf apps
    </dt>
    <dd>
      0
    </dd>
  </dl>
  <dl>
    <dt>
      Deploy times this month
    </dt>
    <dd>
      0
    </dd>
  </dl>
  <dl>
    <dt>
      Total of deploy
    </dt>
    <dd>
      0
    </dd>
  </dl>
</div>
`;
